backend/src/controllers/RestaurantController.ts

```diff
@@ -62,6 +62,21 @@ const searchRestaurants = async (req: Request, res: Response) => {
   }
 };
 
+const getRestaurant = async (req: Request, res: Response) => {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/45#issuecomment-2413380193>#211943</a><span> - 10/15/2024, 09:46 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The getRestaurant function could benefit from additional error handling and input validation.
</div>
<div id="fix">
<b>Fix:</b> Implement input validation for the restaurantId parameter and enhance error handling with more specific error messages.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -65,14 +65,20 @@
  const getRestaurant = async (req: Request, res: Response) => {
    try {
      const restaurantId = req.params.restaurantId;
 +    if (!restaurantId || typeof restaurantId !== 'string' || !restaurantId.match(/^[0-9a-fA-F]{24}$/)) {
 +      return res.status(400).json({ message: 'Invalid restaurant ID' });
 +    }
 +
      const restaurant = await Restaurant.findById(restaurantId).lean();
      if (!restaurant) {
        return res.status(404).json({ message: 'Restaurant not found' });
      }
      res.json(restaurant);
    } catch (error) {
 -    console.log(error);
 -    res.status(500).json({ message: 'Error fetching restaurant' });
 +    console.error('Error in getRestaurant:', error);
 +    res.status(500).json({ message: 'An unexpected error occurred while fetching the restaurant' });
    }
  };
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged